parser/src/lib.rs

```diff
@@ -29,6 +28,51 @@ mod grammar;
 mod infix;
 mod lexer;
 
+/// Shrink hidden spans to fit the visible expressions and flatten singleton blocks.
+fn shrink_hidden_spans<Id>(mut expr: SpannedExpr<Id>) -> SpannedExpr<Id> {
+    match expr.value {
+        Expr::IfElse(_, _, ref last) |
+        Expr::LetBindings(_, ref last) |
+        Expr::TypeBindings(_, ref last) => {
+            let end = last.span.end;
+            drop(last);
```

Comment:
Pretty sure this and the other drops does not do anythingdrop does not do anything.